client/src/services/notificationCenter.ts

```diff
@@ -10,92 +14,123 @@ export enum NotificationLevel {
   Error = 'ERROR',
 }
 
-export interface Notification {
-  id: string,
-  title?: string,
-  message: string,
-  level: NotificationLevel,
-  read: boolean,
-  time: DateTime,
-  data: object,
+export class Notification {
+  id: string;
+  time: DateTime;
+  read: boolean;
+  message: string;
+  level: NotificationLevel;
+  data: object;
+  title?: string;
+  constructor({
+    message,
+    level,
+    data,
+    title,
+  }: {
+    message: string,
+    level: NotificationLevel,
+    data: object,
```

Comment:
`data` should be optional (or have a default value of an empty objet). In the vast majority of cases it won't be used, I just added it in case we have specific information to provide to whatever is using the notification.